cpp/util/etcd_test.cc

```diff
@@ -245,78 +256,106 @@ TEST_F(EtcdTest, TestGetAll) {
 }
 
 TEST_F(EtcdTest, TestCreate) {
-  EXPECT_CALL(client_, Generic(kEntryKey,
-                               AllOf(Contains(Pair(kValueParam, "123")),
-                                     Contains(Pair(kPrevExistParam, kFalse))),
-                               UrlFetcher::Verb::PUT, _, _))
-      .WillOnce(Invoke(
-          bind(&GenericReturn, Status::OK, MakeJson(kCreateJson), 1, _4, _5)));
+  EXPECT_CALL(
+      url_fetcher_,
+      Fetch(IsUrlFetchRequest(
+                UrlFetcher::Verb::PUT, URL(GetEtcdUrl(kEntryKey)),
+                ElementsAre(Pair(StrCaseEq("content-type"),
+                                 "application/x-www-form-urlencoded")),
+                "consistent=true&prevExist=false&quorum=true&value=123"),
+            _, _))
+      .WillOnce(
+          Invoke(bind(HandleFetch, Status::OK, 200,
+                      UrlFetcher::Headers{make_pair("x-etcd-index", "1")},
```

Comment:
I actually ported that straight through (note the 4th parameter of the ```bind()``` that was deleted)...

```EtcdClient``` is kind of inconsistent (at the moment) about what it puts in the ```etcd_index``` field of its ```Response```: sometimes it's the ```X-Etcd-Index``` header, sometimes it's from the Node that it received... I'm being very careful when changing this at the moment, to avoid breaking anything, but my intent is for the ```Response::etcd_index``` to always be the ```X-Etcd-Index```, and if you care about one of the node indexes, just look at them yourself, since we now provide them all (we didn't use to).